crates/jit/src/instantiate.rs

```diff
@@ -116,25 +110,24 @@ impl CompilationArtifacts {
 
         let list = maybe_parallel!(translations.(into_iter | into_par_iter))
             .map(|mut translation| {
+                validate(&translation).map_err(|e| SetupError::Validate(e))?;
+
                 let Compilation {
                     obj,
                     unwind_info,
                     funcs,
                 } = compiler.compile(&mut translation, &types)?;
 
                 let ModuleTranslation {
-                    module,
+                    mut module,
                     data_initializers,
                     debuginfo,
                     has_unparsed_debuginfo,
                     ..
                 } = translation;
 
-                let data_initializers = data_initializers
-                    .into_iter()
-                    .map(OwnedDataInitializer::new)
-                    .collect::<Vec<_>>()
-                    .into_boxed_slice();
+                module.memory_initialization =
+                    Some(MemoryInitialization::new(&module, data_initializers));
```

Comment:
I believe that was the intention behind doing this post-translation originally because the data was stored on `CompiledModule` vs `Module`, but the uffd implementation needs the initialization data on `Module` to access it via `Instance` at runtime (`Instance` currently only references the `Module`).

I would be fine doing it on module translation (make sense to give back a fully-formed `Module`), but I wasn't sure if it was the right thing to do in terms of boxing the initialization data.

Reply:
I think it was originally intended to be a bit more flexible than it is now, but nowadays I think it's best to avoid the `Option` here to be in weird situations where you have to handle both `Some` and `None` somehow. Basically I think it's fine to go ahead and box up the data during module translation. If this is ever a problem in the future we can tackle it then.